config/env/env_types.go

```diff
@@ -0,0 +1,161 @@
+package env
+
+import (
+	"errors"
+	"os"
+	"regexp"
+	"strconv"
+)
+
+var TypeError = errors.New("type could not be converted")
+var EnvironmentError = errors.New("environment variable not found")
+
+var regex *regexp.Regexp
+
+func init() { regex = regexp.MustCompile(`\${[A-Z]+[A-Z1-9_]*}`) }
```

Comment:
let's add a comment around this regex explaining the string it matches to help future consumers of this package. Also does this need to be in an `init()` method, or can the package var `regex` be set directly?